src/shared_realm.cpp

```diff
@@ -607,10 +607,14 @@ void Realm::notify()
             m_coordinator->advance_to_ready(*this);
         }
         else  {
+            // In case the SharedRealm gets descturcted in the did_change callback.
+            auto realm_ref = this->shared_from_this();
```

Comment:
hmmm, that is a good point. Let me check if it is doable to only call close in java `Realm.close()` and delete the `SharedRealm` when it gets GCed.